src/main/scala/symsim/examples/concrete/continuousmaze/ContinuousMaze.scala

```diff
@@ -0,0 +1,139 @@
+package symsim
+package examples.concrete.continuousmaze
+
+import cats.{Eq, Foldable, Monad}
+import cats.kernel.BoundedEnumerable
+
+import org.scalacheck.{Arbitrary, Gen}
+
+import symsim.concrete.Randomized
+
+/**
+ *
+ */
+
+case class MazeState (x: Double, y: Double):
+  override def toString: String = s"[x=$x, y=$y]"
+
+type MazeObservableState = MazeState
+type MazeReward = Double
+
+enum Direction:
+  case R, L, U, D
+
+enum Velocity:
+  case B, S
+
+type MazeAction = (Direction, Velocity)
+
+val MAZE_LENGTH: Int = 4
+val MAZE_WIDTH: Int = 3
+
+val BIG_STEP: Double = 1
+val SMALL_STEP: Double = 0.5
+
+object ContinuousMaze
+  extends 
+    Agent[MazeState, MazeObservableState, MazeAction, MazeReward, Randomized],
+    Episodic:
+
+  val TimeHorizon: Int = 2000
+
+  def isFinal (s: MazeState): Boolean =
+    s == MazeState (MAZE_LENGTH, MAZE_WIDTH) || s == MazeState (MAZE_LENGTH, MAZE_WIDTH - 1)
```

Comment:
It seems that this is trying to hit a particular point in the continuous rectangle, but we need to hit a square inside that rectangle. Otherwise, IMHO, the problem is too difficult for what we need.

Reply:
According to the agent class, it should be maze state.

Reply:
Then, I think the code is incorrect either here or there. If `isFinal` is called on observable state, then the set of the final states should be an entire square, not just two points. If `isFinal` is called by symsim on observable state, then they should require the observable state type, not the system state type? Either side is confused. It does not make sense to check an observable state condition on a continuous state ...

Reply:
> Then, I think the code is incorrect either here or there. If `isFinal` is called on observable state, then the set of the final states should be an entire square, not just two points. If `isFinal` is called by symsim on observable state, then they should require the observable state type, not the system state type? Either side is confused. It does not make sense to check an observable state condition on a continuous state ...

Maybe I wasn't clear, I meant it is defined on `MazeState`. If it was defined on `MazeObservableState`, you were right.